enterprise/internal/executor/client_types.go

```diff
@@ -97,6 +97,16 @@ type MarkErroredRequest struct {
 type HeartbeatRequest struct {
 	ExecutorName string `json:"executorName"`
 	JobIDs       []int  `json:"jobIds"`
+
+	// Telemetry data.
+
+	OS            string `json:"os"`
+	Architecture  string `json:"architecture"`
+	SrcCliVersion string `json:"srcCliVersion"`
+	GitVersion    string `json:"gitVersion"`
+	DockerVersion string `json:"dockerVersion"`
+	IgniteVersion string `json:"igniteVersion"`
+	Version       string `json:"version"`
```

Comment:
```suggestion
	QueueName       string `json:"queueName"`
	OS              string `json:"os"`
	Architecture    string `json:"architecture"`
	ExecutorVersion string `json:"executorVersion"`
	SrcCliVersion   string `json:"srcCliVersion"`
	GitVersion      string `json:"gitVersion"`
	DockerVersion   string `json:"dockerVersion"`
	IgniteVersion   string `json:"igniteVersion"`
```

Here's the fields I have in the frontend-facing change:

Reply:
Version might be more specific as ExecutorVersion, and we're missing QueueName.

Reply:
I dropped QueueName as it's in the URL path already